kuka_rsi_driver/src/robot_manager_node_eki_rsi.cpp

```diff
@@ -0,0 +1,365 @@
+
+  // Publish initial control mode
+  std_msgs::msg::UInt32 message;
+  message.data = static_cast<int>(control_mode_);
+  control_mode_pub_->publish(message);
+
+  // Configure hardware interface
+  const bool hw_state_change_successful = kuka_drivers_core::changeHardwareState(
+    change_hardware_state_client_, robot_model_, State::PRIMARY_STATE_INACTIVE);
+  if (!hw_state_change_successful)
+  {
+    RCLCPP_ERROR(logger, "Could not configure hardware interface");
+    return FAILURE;
+  }
+
+  // Activate control mode handler and event broadcaster controllers
+  std::vector<std::string> controllers_to_activate{
+    kuka_drivers_core::CONTROL_MODE_HANDLER,
+    kuka_drivers_core::NRT_MESSAGE_HANDLER,
+    kuka_drivers_core::EVENT_BROADCASTER,
+  };
+  const bool controller_activation_successful = kuka_drivers_core::changeControllerState(
+    change_controller_state_client_, controllers_to_activate, {});
+  if (!controller_activation_successful)
+  {
+    RCLCPP_ERROR(logger, "Could not activate control mode handler or event broadcaster controller");
+    on_cleanup(get_current_state());
+    return FAILURE;
+  }
+
+  // Publish that the hardware interface is now configured
+  // This will start the control loop
+  is_configured_pub_->on_activate();
+  is_configured_msg_.data = true;
+  is_configured_pub_->publish(is_configured_msg_);
+
+  return SUCCESS;
+}
+
+CallbackReturn RobotManagerNodeEkiRsi::on_cleanup(const rclcpp_lifecycle::State &)
+{
+  const auto logger = get_logger();
+
+  // Deactivate control mode handler and event broadcaster
+  std::vector<std::string> controllers_to_deactivate{
+    kuka_drivers_core::CONTROL_MODE_HANDLER,
+    kuka_drivers_core::NRT_MESSAGE_HANDLER,
+    kuka_drivers_core::EVENT_BROADCASTER,
+  };
+  const bool controller_deactivation_successful = kuka_drivers_core::changeControllerState(
+    change_controller_state_client_, {}, controllers_to_deactivate);
+  if (!controller_deactivation_successful)
+  {
+    RCLCPP_ERROR(logger, "Could not deactivate control mode handler and event broadcaster");
+  }
+
+  // Clean up hardware interface
+  const bool hw_state_change_successful = kuka_drivers_core::changeHardwareState(
+    change_hardware_state_client_, robot_model_, State::PRIMARY_STATE_UNCONFIGURED);
+  if (!hw_state_change_successful)
+  {
+    RCLCPP_ERROR(logger, "Could not clean up hardware interface");
+    return FAILURE;
+  }
+
+  if (is_configured_pub_->is_activated())
+  {
+    is_configured_msg_.data = false;
+    is_configured_pub_->publish(is_configured_msg_);
+    is_configured_pub_->on_deactivate();
+  }
+
+  return SUCCESS;
+}
+
+CallbackReturn RobotManagerNodeEkiRsi::on_activate(const rclcpp_lifecycle::State &)
+{
+  const auto logger = get_logger();
+  terminate_ = false;
+
+  // Activate hardware interface
+  const bool hw_state_change_successful = kuka_drivers_core::changeHardwareState(
+    change_hardware_state_client_, robot_model_, State::PRIMARY_STATE_ACTIVE, 10'000);
```

Comment:
10 sec might be enough, but there is a 10sec timeout in the hardware interface activation, when waiting for drives turning on, therefore I would increase this one  a little higher (or lower that one from 10)